### ISSUE TITLE
 Bug: Gemini function call not processing correctly, resulting in TypeError

### ISSUE THREAD

**Issue by minhquan23102000** [Mirascope/mirascope#652]
Expected behavior: The Gemini function call should successfully process and append the output message of tools to the history without errors.

Actual behavior: When attempting to append the output message, a TypeError occurs, indicating that the expected type is not being met. The following error message is displayed:
```
TypeError: Could not create `Blob`, expected `Blob`, `dict` or an `Image` type(`PIL.Image.Image` or `IPython.display.Image`). Got a: <class 'google.ai.generativelanguage_v1beta.types.content.Content'> Value: parts { text: "Hello, AI!" } role: "user"
```

Steps to reproduce:

1. Execute the function call `get_weather_in_tokyo` with the input `Hello, AI!`.
2. Observe the response and attempt to append the output message to the history.
3. Note the TypeError that is raised during this process.

Code reproduce:
```python
from mirascope.core import litellm, BaseMessageParam, prompt_template, gemini
from rich import print

history: list[BaseMessageParam | gemini.GeminiCallResponse] = [
    BaseMessageParam(role="user", content="Hello, AI!"),
    BaseMessageParam(role="agent 001", content="Hello, user!"),
    BaseMessageParam(role="user", content="What is the weather in Tokyo?"),
    BaseMessageParam(
        role="agent 001", content="User asked about the weather in Tokyo."
    ),
]


def get_weather_in_tokyo(city: str) -> str:
    """Get the weather in a city"""
    return f"The weather in {city} is 20 degrees Celsius."


@gemini.call(model="gemini-1.5-flash-002", tools=[get_weather_in_tokyo])
@prompt_template(
    """
    MESSAGES: {history}
    """
)
def test(history): ...


response = test(history)

print(response.content)

history.append(response.message_param)

if response.tool:
    tool_output = response.tool.call()
    history.append(response.tool_message_params([(response.tool, tool_output)]))

    print(history)


new_response = test(history)
print(new_response.content)
```

Code output: 
```bash
 py .\test.py
[
    BaseMessageParam(role='user', content='Hello, AI!'),
    BaseMessageParam(role='agent 001', content='Hello, user!'),
    BaseMessageParam(role='user', content='What is the weather in Tokyo?'),
    BaseMessageParam(role='agent 001', content='User asked about the weather in Tokyo.'),
    {'role': 'model', 'parts': [function_call {
  name: "get_weather_in_tokyo"
  args {
    fields {
      key: "city"
      value {
        string_value: "tokyo"
      }
    }
  }
}
]},
    [name: "get_weather_in_tokyo"
response {
  fields {
    key: "result"
    value {
      string_value: "The weather in tokyo is 20 degrees Celsius."
    }
  }
}
]
]
Traceback (most recent call last):
  File "D:\Work\personal\personal-agent\test.py", line 39, in <module>
    new_response = test(history)
                   ^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\mirascope\core\base\_create.py", line 212, in inner
    response = create(stream=False, **call_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\mirascope\core\base\_utils\_get_create_fn_or_async_create_fn.py", line 134, in create_or_stream
    return cast(_NonStreamedResponse, sync_func(**kwargs))
                                      ^^^^^^^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\google\generativeai\generative_models.py", line 305, in generate_content
    request = self._prepare_request(
              ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\google\generativeai\generative_models.py", line 154, in _prepare_request
    contents = content_types.to_contents(contents)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\google\generativeai\types\content_types.py", line 333, in to_contents
    contents = [to_content(contents)]
                ^^^^^^^^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\google\generativeai\types\content_types.py", line 296, in to_content
    return protos.Content(parts=[to_part(part) for part in content])
                                 ^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\google\generativeai\types\content_types.py", line 264, in to_part
    return protos.Part(inline_data=to_blob(part))
                                   ^^^^^^^^^^^^^
  File "D:\Work\personal\personal-agent\.venv\Lib\site-packages\google\generativeai\types\content_types.py", line 210, in to_blob
    raise TypeError(
TypeError: Could not create `Blob`, expected `Blob`, `dict` or an `Image` type(`PIL.Image.Image` or `IPython.display.Image`).
Got a: <class 'google.ai.generativelanguage_v1beta.types.content.Content'>
Value: parts {
  text: "Hello, AI!"
}
role: "user"

```

Environment:

- Operating System: Windows 11
- Python version: 3.12
- Library version: 
```bash
mirascope                    1.6.1
google-ai-generativelanguage 0.6.10
google-api-core              2.22.0
google-api-python-client     2.149.0
google-auth                  2.35.0
google-auth-httplib2         0.2.0
google-generativeai          0.8.3
```

### 

Confirm the code run ok if without the tools

**Comment by willbakst** [comment]
Release fixed in `v1.7.1`!!

**Comment by willbakst** [comment]
Ah sorry, I forgot to include necessary code updates for Gemini:

```python
from rich import print

from mirascope.core import BaseMessageParam, gemini, prompt_template

# UPDATED
history: list[gemini.GeminiMessageParam] = [
    BaseMessageParam(role="user", content="Hello, AI"),
    BaseMessageParam(role="assistant", content="Hello, user!"),
    BaseMessageParam(role="user", content="What is the weather in Tokyo"),
]

# INCORRECT (Gemini needs to have alternating user / assistant message chain, with an optional initial (pseudo) system message)
#  history: list[BaseMessageParam | gemini.GeminiCallResponse] = [
#      BaseMessageParam(role="user", content="Hello, AI!"),
#      BaseMessageParam(role="agent 001", content="Hello, user!"),
#      BaseMessageParam(role="user", content="What is the weather in Tokyo?"),
#      BaseMessageParam(
#          role="agent 001", content="User asked about the weather in Tokyo."
#      ),
#  ]


def get_weather_in_tokyo(city: str) -> str:
    """Get the weather in a city"""
    return f"The weather in {city} is 20 degrees Celsius."


@gemini.call(model="gemini-1.5-flash-002", tools=[get_weather_in_tokyo])
@prompt_template(
    """
    MESSAGES: {history}
    """
)
def test(history): ...


response = test(history)
history.append(response.message_param)

if tool := response.tool:
    # UPDATED
    history += response.tool_message_params([(tool, tool.call())])
    # INCORRECT (`tool_message_params` returns a list)
    # history.append(response.tool_message_params([(response.tool, tool_output)]))

new_response = test(history)
print(new_response.content)

```
